webapp/src/pages/ChatRoom/Messages.tsx

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useState } from 'react';
+import { ChatBubble, ChatBubbleVariant } from '../../shared/components';
+import { User } from '../../shared/generated';
+import { useAuth } from '../../shared/hooks/UseAuth';
+import socket from '../../shared/socket';
+import { AVATAR_EP_URL, WILDCARD_AVATAR_URL } from '../../shared/urls';
+
+type MessageType = {
+  id: string;
+  user: User;
+  content: string;
+  createdAt: number;
+};
+
+function Messages() {
+  const [messages, setMessages] = useState<MessageType[]>([]);
+  const [connectedUsers, setConnectedUsers] = useState<Map<string, User>>(
+    new Map<string, User>(),
+  );
+  const { user: me } = useAuth();
+
+  useEffect(() => {
+    const messagesListener = (messages: MessageType[]) => {
+      setMessages(messages);
+    };
+
+    const messageListener = (message: MessageType) => {
+      setMessages((prevMessages) => {
+        const newMessages = [...prevMessages, message];
```

Comment:
I understand this is somehow a draft, because we are not persisting the messages ye4t, but I assume when we do, this whole logic to display older messages will become simpler?

Reply:
Here we use an [updater function](https://beta.reactjs.org/apis/react/useState#examples-updater) to skip adding the `messages` as a dependency of the `useEffect` hook.

When we add persistence, this will stay mostly the same because we want to display the previous messages and add the new message (if we figure out another way, we can change it, of course).

Reply:
What about users that join the chat later on? Or users that join, leave, and then rejoin? Shouldn't they be able to see all chats since the beginning? Wouldn't this have to persist in the backend?

Reply:
All of them, yes.